libnmstate/nm/bond.py

```diff
@@ -79,14 +80,23 @@ def _get_options(nm_device):
         bond_option_names_in_profile.add("arp_interval")
         bond_option_names_in_profile.add("miimon")
 
+    # Mode is required
+    sysfs_folder = SYSFS_BOND_OPTION_FOLDER_FMT.format(ifname=ifname)
+    mode = _read_sysfs_file(f"{sysfs_folder}/mode")
+    if not mode:
+        return {}
+
+    bond_setting = nmclient.NM.SettingBond.new()
+    bond_setting.add_option("mode", mode)
+
     options = {}
     for sysfs_file in glob.iglob(f"/sys/class/net/{ifname}/bonding/*"):
         option = os.path.basename(sysfs_file)
         if option in NM_SUPPORTED_BOND_OPTIONS:
             value = _read_sysfs_file(sysfs_file)
+            default_value = bond_setting.get_option_default(option)
             if (
-                option == "mode"
-                or value != bond_setting.get_option_default(option)
+                (default_value and value != default_value)
```

Comment:
We always want to report the `mode`, even if the current one is the default.
Won't this change cause it not to be reported?

Reply:
Add the `mode` explicitly.